lib/modules/datasource/unity3d/index.spec.ts

```diff
@@ -0,0 +1,233 @@
+import { getPkgReleases } from '..';
+import { Fixtures } from '../../../../test/fixtures';
+import * as httpMock from '../../../../test/http-mock';
+import { Unity3dDatasource } from '.';
+
+const mockRSSFeeds = (streams: { [keys: string]: string }) => {
+  Object.entries(streams).map(([stream, url]) => {
+    const content = Fixtures.get(stream + '.xml');
+
+    const uri = new URL(url);
+
+    httpMock.scope(uri.origin).get(uri.pathname).reply(200, content);
+  });
+};
+
+describe('modules/datasource/unity3d/index', () => {
+  it('handle 500 response', async () => {
+    const uri = new URL(Unity3dDatasource.streams.stable);
+    httpMock.scope(uri.origin).get(uri.pathname).reply(500, '500');
+
+    const qualifyingStreams = { ...Unity3dDatasource.streams };
+    delete qualifyingStreams.beta;
+    const response = await getPkgReleases({
+      datasource: Unity3dDatasource.id,
+      packageName: 'm_EditorVersion',
+      registryUrls: [Unity3dDatasource.streams.stable],
+    });
+
+    expect(response).toBeNull();
+  });
+
+  it('handle 200 with no XML', async () => {
+    const uri = new URL(Unity3dDatasource.streams.stable);
+    httpMock.scope(uri.origin).get(uri.pathname).reply(200, 'not xml');
+
+    const qualifyingStreams = { ...Unity3dDatasource.streams };
+    delete qualifyingStreams.beta;
+    const response = await getPkgReleases({
+      datasource: Unity3dDatasource.id,
+      packageName: 'm_EditorVersion',
+      registryUrls: [Unity3dDatasource.streams.stable],
+    });
+
+    expect(response).toBeNull();
+  });
+
+  it('handles missing title element', async () => {
+    const content = Fixtures.get('no_title.xml');
+    const uri = new URL(Unity3dDatasource.streams.stable);
+    httpMock.scope(uri.origin).get(uri.pathname).reply(200, content);
+
+    const qualifyingStreams = { ...Unity3dDatasource.streams };
+    delete qualifyingStreams.beta;
+    const responses = await getPkgReleases({
+      datasource: Unity3dDatasource.id,
+      packageName: 'm_EditorVersion',
+      registryUrls: [Unity3dDatasource.streams.stable],
+    });
+
+    expect(responses).toEqual(
+      expect.objectContaining({
+        releases: expect.arrayContaining([]),
+      }),
+    );
```

Comment:
```suggestion
    expect(responses).toEqual({
      releases: [],
      homepage: 'https://unity.com/',
      registryUrl: 'https://unity.com//releases/editor/releases.xml',
    });
```
This is what I have meant with use `toEqual`, else we do not see if e.g. `registryUrl` is missing.

At least once there should an Equal with all elements, outside from that you can use `toMatchObject` which allows to drop the elements you are not testing. 

Please change the other tests accordingly

Reply:
Adjusted all tests inside ae263ebdc02cf3e3c1020cb743121b26ec7fe2cf and 00b471fe6f624622e27a23bd924670852352b4ac:
- while it worked before, [tests now actually assert that beta stream releases mark `isStable` as `false`](https://github.com/renovatebot/renovate/commit/ae263ebdc02cf3e3c1020cb743121b26ec7fe2cf#diff-a879aa4ba35047dd4c0db2beedcd28fc21da15168c2ee04d7f959d22658f8008R141)
- [tests assert a non-beta and beta-release with all properties](https://github.com/renovatebot/renovate/commit/ae263ebdc02cf3e3c1020cb743121b26ec7fe2cf#diff-a879aa4ba35047dd4c0db2beedcd28fc21da15168c2ee04d7f959d22658f8008L123-R176)
- [test assert the full `getPkgReleases()` response exactly](https://github.com/renovatebot/renovate/commit/00b471fe6f624622e27a23bd924670852352b4ac)

This was a great review point; I've discovered and resolved two different issues as part of testing all properties at least once, so thanks for pointing it out!
- [no more double `//`](https://github.com/renovatebot/renovate/commit/ae263ebdc02cf3e3c1020cb743121b26ec7fe2cf#diff-a3ea6fb68bcf65131e21af23affe109becae232c834f5a1c7e409ca2c700651cL11-R13) (`https://unity3d.com//releases/…` -> `https://unity3d.com/releases/…`)
- [adds `registryUrl` that was missing if `getByStream` returned early because of a parsing error](https://github.com/renovatebot/renovate/pull/27971/commits/ae263ebdc02cf3e3c1020cb743121b26ec7fe2cf#diff-a3ea6fb68bcf65131e21af23affe109becae232c834f5a1c7e409ca2c700651cL49-R53)